src/main/java/com/example/userManagement/controller/UserController.java

```diff
@@ -0,0 +1,155 @@
+     * @return This is a user info!. (status code 200)
+     * or Error Occurred (status code 400)
+     * or Error Occurred (status code 401)
+     * or Error Occurred (status code 404)
+     */
+    @ApiOperation(value = "Get all User data  with some filters.", nickname = "getAllUsers", notes = "Get list of all Users data ", response = UserResponse.class, tags = {"Filter user data",})
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "This is a user info!.", response = UserResponse.class),
+            @ApiResponse(code = 400, message = "Error Occurred", response = GetErrorResponse.class),
+            @ApiResponse(code = 401, message = "Error Occurred", response = GetErrorResponse.class),
+            @ApiResponse(code = 404, message = "Error Occurred", response = GetErrorResponse.class)})
+    @GetMapping(value = "/user")
+    public List<UserIndexModel> getAllUsers(@ApiParam(value = "Query criteria. Refer following links for more details:<br/> - [match](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query.html)<br/> - [multi_match](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html)<br/> - [term](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-term-query.html)<br/> - [terms](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-query.html)<br/> - [range](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html)<br/> - [regexp](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-regexp-query.html)") @Valid Query query) {
+        return userServiceES.findUserByQuery(query);
+    }
+
+
+    /**
+     * GET /user/{userId} : Get user by user id
+     *
+     * @param userId (required)
+     * @return This is a user info!. (status code 200)
+     * or Error Occurred (status code 400)
+     * or Error Occurred (status code 404)
+     */
+    @ApiOperation(value = "Get user by user id", nickname = "getUserById", notes = "", response = User.class, tags = {"user operations",})
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "This is a user info!.", response = User.class),
+            @ApiResponse(code = 400, message = "Error Occurred", response = GetErrorResponse.class),
+            @ApiResponse(code = 404, message = "Error Occurred", response = GetErrorResponse.class)})
+    @GetMapping(
+            value = "/user/{userId}",
+            produces = {"application/json"}
+    )
+    public User getUserById(@ApiParam(value = "", required = true) @PathVariable("userId") UUID userId) {
```

Comment:
What is the purpose of @ApiParam annotation here?

Reply:
I changed the value of API param.